File: AlcoolouGasolina/app/src/main/java/com/example/alcoolougasolina/MainActivity.java
package com.example.alcoolougasolina;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.google.android.material.textfield.TextInputEditText;

public class MainActivity extends AppCompatActivity {
    private TextInputEditText edtGasolina, edtAlcool;
    private TextView txtResult;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edtGasolina = findViewById(R.id.edtGasolina);
        edtAlcool   = findViewById(R.id.edtAlcool);
        txtResult   = findViewById(R.id.txtResult);
    }

    public void calcularPreco(View view){
        //recupera valores digitados
        String gasolina = edtGasolina.getText().toString();
        String alcool = edtAlcool.getText().toString();

        //validar os campos digitados
        Boolean resultado = validarCampos(alcool, gasolina);
        if( resultado ){
            //Convertendo string para números
            Double valorAlcool = Double.parseDouble(alcool);
            Double valorGasolina = Double.parseDouble(gasolina);

            if ((valorAlcool / valorGasolina) >= 0.7) {
                txtResult.setText("Melhor usar gasolina");
            } else {
                txtResult.setText("Melhor utilizar álcool");
            }
        } else {
            txtResult.setText("Preencha os preços primeiro!");
        }
    }

    public Boolean validarCampos(String pAlcool, String pGasolina){
        Boolean camposValidados = true;
        if(pAlcool == null || pAlcool.equals("")){
            camposValidados = false;
        } else if (pGasolina == null || pGasolina.equals("")){
            camposValidados = false;
        }
        return camposValidados;
    }
}

File: AppSorteioKotlin/app/src/main/java/com/caiodetore/appsorteiokotlin/MainActivity.kt
package com.caiodetore.appsorteiokotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.TextView
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val btnCalcular = findViewById<Button>(R.id.button2)
        btnCalcular.setOnClickListener({
            calcularPreco()
        })
    }

    fun calcularPreco(){
        val precoAlcool = edtAlcool.text.toString()
        val precoGasolina = edtGasolina.text.toString()

        val validarCampos = validarCampos(precoAlcool, precoGasolina)
    }

    fun calcularMelhorPreco(precoAlcool: String, precoGasolina: String){
        val valorAlcool = precoAlcool.toDouble()
        val valorGasolina = precoGasolina.toDouble()

        val resultadoPreco = valorAlcool/valorGasolina

        if ( resultadoPreco >= 0.7 ) {
            txtResult.text = "Gasolina >"
        } else {
            txtResult.text = "Alcool >"
        }
    }

    fun validarCampos(precoAlcool: String, precoGasolina: String) : Boolean {

        var camposValidados: Boolean = true;
        if( precoAlcool == null || precoAlcool.equals("") ) {
            camposValidados = false
        } else if ( precoGasolina == null || precoGasolina.equals("") ){
            camposValidados = false
        }

        return camposValidados
    }

    fun sortearNumero(view: View){
        var texto = findViewById<TextView>(R.id.textView)
        var numSort = Random().nextInt(11)
        texto.text = "Número sorteado: " + numSort
    }

}